packages/react-components/lib/systems-summary/systems-summary-item-state.tsx

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import { makeStyles, Typography, Grid, Paper, Button } from '@material-ui/core';
+import NavigateNextIcon from '@material-ui/icons/NavigateNext';
+import { SpoiltRobot, SpoiltDispenser, SpoiltDoor, SpoiltLift } from './index';
+
+export interface ItemSummaryState {
+  operational: number;
+  outOfOrder: number;
+}
+
+export interface RobotSummaryState extends ItemSummaryState {
+  idle: number;
+  charging: number;
+}
+
+export interface ItemSummary {
+  item: string;
+  itemSummary?: ItemSummaryState;
+  robotSummary?: RobotSummaryState;
+}
+
+export interface DoorSummary extends ItemSummary {
+  spoiltDoors: SpoiltDoor[];
+}
+
+export interface LiftSummary extends ItemSummary {
+  spoiltLifts: SpoiltLift[];
+}
+
+export interface DispenserSummary extends ItemSummary {
+  // list of dispenser names
+  spoiltDispensers: SpoiltDispenser[];
+}
+
+export interface RobotSummary extends ItemSummary {
+  spoiltRobots: SpoiltRobot[];
+}
+
+export interface SystemSummaryItemStateProps {
+  itemSummary: ItemSummary;
+  onClick: () => void;
+}
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    padding: theme.spacing(0.2),
+  },
+  warning: {
+    color: theme.palette.error.main,
+    border: `2px solid ${theme.palette.error.main}`,
+  },
+  operational: {
+    color: theme.palette.success.main,
+    border: `2px solid ${theme.palette.success.main}`,
+  },
+  idle: {
+    color: theme.palette.warning.main,
+    border: `2px solid ${theme.palette.warning.main}`,
+  },
+  charging: {
+    color: theme.palette.info.main,
+    border: `2px solid ${theme.palette.info.main}`,
+  },
+  headerGrid: {
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+  button: {
+    padding: 0,
+  },
+  robotGrid: {
+    marginTop: theme.spacing(1),
+  },
+}));
+
+export const SystemSummaryItemState = (props: SystemSummaryItemStateProps): JSX.Element => {
+  const classes = useStyles();
+  const { itemSummary, onClick } = props;
+
+  let totalItem = 0;
+  let operationalItem = 0;
+
+  if (itemSummary.itemSummary) {
+    totalItem = itemSummary.itemSummary.operational + itemSummary.itemSummary.outOfOrder;
+    operationalItem = itemSummary.itemSummary.operational;
+  } else if (itemSummary.robotSummary) {
+    totalItem = itemSummary.robotSummary.operational + itemSummary.robotSummary.outOfOrder;
+    operationalItem = itemSummary.robotSummary.operational;
+  }
```

Comment:
Since robot states is different from other states, I think it's better to separate them into different components instead of changing behavior based on the type of the props.

This is also not strictly type safe as `itemSummary` and `robotSummary` could both be undefined.